syncd/BestCandidateFinder.cpp

```diff
@@ -1886,6 +1886,9 @@ std::shared_ptr<SaiObj> BestCandidateFinder::findCurrentBestMatchForGenericObjec
                 soci.equal_attributes);
 
         candidateObjects.push_back(soci);
+        if (soci.equal_attributes == attrs.size()
+            && (object_type == SAI_OBJECT_TYPE_VLAN || object_type == SAI_OBJECT_TYPE_VLAN_MEMBER))
```

Comment:
This is object specific, please remove it from this place and add object specific logic for finding best candidatte